crates/sui-core/src/authority/authority_store.rs

```diff
@@ -338,17 +259,19 @@ impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
         }
     }
 
+    #[allow(clippy::type_complexity)]
     pub fn get_sequenced_input_objects(
         &self,
         digest: &TransactionDigest,
         objects: &[InputObjectKind],
-    ) -> Result<Vec<Object>, SuiError> {
+    ) -> Result<(Vec<Object>, Vec<ObjectKey>, Vec<SuiError>), SuiError> {
```

Comment:
We should add a new function to return the missing objects, so that we can avoid reading and deserializing objects from withing TransactionManager. Also, this function is already very hard to understand, and a frequent cause of confusion and bugs due to the errors that it returns, so I would strongly prefer not to add any new complication here (even if that means we have to duplicate a small amount of code).